docs/userguides/contracts.md

```diff
@@ -256,6 +256,24 @@ If you need the `return_value` from a transaction, you have to either treat tran
 assert receipt.return_value == 123
 ```
 
+Transactions may also fail.
+When a transaction fails, Ape (by default) raises the virtual machine error which may crash your script.
+You will see exception trace data like this:
+
+```
+  File "$HOME/ApeProjects/ape-playground/scripts/fail.py", line 7, in main
+    contract.setNumber(5, sender=owner)
+
+ERROR: (ContractLogicError) Transaction failed.
+```
+
+If you know a transaction is going to revert and you wish to allow it, use the `fail_on_revert=False` flag:
+
+```python
+receipt = contract.setNumber(5, sender=owner, fail_on_revert=False)
+assert receipt.failed
+```
```

Comment:
Might be worth showing what happens when you try to use certain properties of the receipt that aren't available when it fails e.g. `.return_value`

Reply:
you know what it is weird (or cool), i think `.return_value` is the revert message or something:

```
>>> receipt.return_vaue
'0x08c379a00000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000000b21617574686f72697a6564000000000000000000000000000000000000000000'
```

Reply:
i think in that case, the receipt data was available though.

Reply:
that was when i used a geth node though. Using `eth-tester`, i get an empty tuple!

Reply:
i guess what im saying is that the properties will always be available in one way or another

Reply:
I think we can either document it as "the value of `.return_value` when `.failed` is true is undocumented behavior", or standardizing what `.return_value` is for all the different providers (1 sounds better)

Reply:
oh that probably only has to do with tracing, since eth-tester doesnt have tracing and the receipt data doesnt contain the return value (I think Charles has a EIP for this)